pkg/search/proxy/store/multi_cluster_cache.go

```diff
@@ -334,15 +334,44 @@ func (c *MultiClusterCache) Watch(ctx context.Context, gvr schema.GroupVersionRe
 		if cache == nil {
 			continue
 		}
-		w, err := cache.Watch(ctx, options)
-		if err != nil {
+
+		// The following logic adds a 30-second timeout to prevent watch requests to member clusters from hanging,
+		// which could cause the client watch to hang.
+		watchChan := make(chan watch.Interface, 1)
+		errChan := make(chan error, 1)
+
+		go func(cluster string) {
+			w, err := cache.Watch(ctx, options)
```

Comment:
If the watcher times out after 30 seconds during creation, it will trigger a time.After timeout, return an error, and call cancel to stop the watcher goroutine.

```
https://github.com/karmada-io/karmada/blob/e65e993d2d0cf920b889daa7f5c36e94d6829787/vendor/k8s.io/apiserver/pkg/endpoints/handlers/get.go#L263
```